src/Microsoft.ML.TimeSeries/ExtensionsCatalog.cs

```diff
@@ -187,5 +187,33 @@ public static class TimeSeriesCatalog
             new SsaForecastingEstimator(CatalogUtils.GetEnvironment(catalog), outputColumnName, inputColumnName, windowSize, seriesLength, trainSize,
                 horizon, isAdaptive, discountFactor, rankSelectionMethod, rank, maxRank, shouldStabilize, shouldMaintainInfo, maxGrowth, confidenceLowerBoundColumn,
                 confidenceUpperBoundColumn, confidenceLevel, variableHorizon);
+
+        /// <summary>
+        /// Create <see cref="SrCnnEntireAnomalyEstimator"/>, which detects timeseries anomalies for entire input using SRCNN algorithm.
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.
+        /// The column data is a vector of <see cref="System.Double"/>. The length of this vector varies depending on <paramref name="srCnnDetectMode"/>.</param>
+        /// <param name="inputColumnName">Name of column to transform. The column data must be <see cref="SrCnnTsPoint"/>.</param>
+        /// <param name="threshold">The threshold to determine anomaly, score larger than the threshold is considered as anomaly. Must be in [0,1]. Default value is 0.3.</param>
+        /// <param name="batchSize">Divide the input data into batches to fit srcnn model.
+        /// When set to -1, use the whole input to fit model instead of batch by batch, when set to a positive integer, use this number as batch size.
```

Comment:
It appears that it is the batch size that is driving this new estimator. 

Would it be better to call this `SrCnnBatchedAnomalyEstimator` and the function as `DetectAnomalySrCnnInBatchedMode` ?

Reply:
The key difference of this new estimator with the old one is that it implements entire anomaly detection for timeseries, instead of streaming anomaly detection in the old one. The batch size here is an optional parameter. When the timeseries is too long, it's better to divide the timeseries into batches to do entire anomaly detection to get better performance and higher quality.

Reply:
> It appears that it is the batch size that is driving this new estimator.
> 
> Would it be better to call this `SrCnnBatchedAnomalyEstimator` and the function as `DetectAnomalySrCnnInBatchedMode` ?

The key difference of this new estimator with the old one is that it implements entire anomaly detection for timeseries, instead of streaming anomaly detection in the old one. The batch size here is an optional parameter. When the timeseries is too long, it's better to divide the timeseries into batches to do entire anomaly detection to get better performance and higher quality.

Reply:
If a batch value is provided then the detection does not happen on the entire series at once, but happens in batches. But with a batch size of -1, the detection happens on the whole series. 
Even though batch size is optional with a default value, it is this parameter that controls the behavior.  So I would like to suggest a name change as above (`SrCnnBatchedAnomalyEstimator` and the function as `DetectAnomalySrCnnInBatchedMode`)

---
In reply to: [416644152](https://github.com/dotnet/machinelearning/pull/4998#discussion_r416644152) [](ancestors = 416644152)

Reply:
> If a batch value is provided then the detection does not happen on the entire series at once, but happens in batches. But with a batch size of -1, the detection happens on the whole series.
> Even though batch size is optional with a default value, it is this parameter that controls the behavior. So I would like to suggest a name change as above (`SrCnnBatchedAnomalyEstimator` and the function as `DetectAnomalySrCnnInBatchedMode`)
> 
> In reply to: [416644152](https://github.com/dotnet/machinelearning/pull/4998#discussion_r416644152) [](ancestors = 416644152)

I have some concerns to use "batch", since firstly, there is already a sample called DetectAnomalyBySrCnnBatchPrediction(https://github.com/dotnet/machinelearning/blob/master/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectAnomalyBySrCnnBatchPrediction.cs), this sample is for the old streaming estimator, but call it in a "batch" way. This sample was added following other models of TimeSeries. I'm not sure if it will confuse users if we call the new estimator "batch". And besides, in Kensho Anomaly Detector API, we call the two APIs "Last API" and "Entire API", so it would be better if we could continue use the name "Entire" to make it easier to switch among different platforms (also we are relasing SR to AML and uses the same word).

Reply:
Sounds good. Thanks for the explanation.

---
In reply to: [418108643](https://github.com/dotnet/machinelearning/pull/4998#discussion_r418108643) [](ancestors = 418108643)